### PR TITLE
[Dataset] Builtin knowledge graph dataset

### PR DESCRIPTION
**Pull request by classicsong**
## Description
Builtin dataset of FB15k, FB15k-237, WN18

#1776 
#1629 

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR

### CODE REVIEW

File: python/dgl/data/knowledge_graph.py [hunk 1]
```diff
@@ -0,0 +1,737 @@
...
+
+    def download(self):
+        r""" Automatically download data and extract it.
+        """
+        tgz_path = os.path.join(self.raw_dir, self.name + '.tgz')
+        download(self.url, path=tgz_path)
+        extract_archive(tgz_path, self.raw_path)
+
+    def process(self):
+        """
```
Comment (hetong007) [inline python/dgl/data/knowledge_graph.py:60]: I'm a bit confused. what is the purpose of the list of params?

File: python/dgl/data/knowledge_graph.py [hunk 2]
```diff
@@ -0,0 +1,737 @@
...
+        r"""Gets the graph object
+
+        Parameters
+        -----------
+        idx: int
+            Item index, FB15k237Dataset has only one graph object
+
+        Return
+        -------
+            dgl.DGLGraph
```
Comment (hetong007) [inline python/dgl/data/knowledge_graph.py:456]: Probably wrong indent

File: python/dgl/data/knowledge_graph.py [hunk 3]
```diff
@@ -0,0 +1,737 @@
...
+        r"""Gets the graph object
+
+        Parameters
+        -----------
+        idx: int
+            Item index, FB15kDataset has only one graph object
+
+        Return
+        -------
+            dgl.DGLGraph
```
Comment (hetong007) [inline python/dgl/data/knowledge_graph.py:579]: same indent issue

File: python/dgl/data/knowledge_graph.py (hunk 1 above)
Comment (classicsong) [reply python/dgl/data/knowledge_graph.py:60]: I think I need to change the description.

**Pull request merged**